pandas/core/arrays/datetimelike.py

```diff
@@ -1427,8 +1435,10 @@ def _time_shift(
         if self.freq is None:
             raise NullFrequencyError("Cannot shift with no freq")
 
-        start = self[0] + periods * self.freq
-        end = self[-1] + periods * self.freq
+        # error: No overload variant of "__radd__" of "BaseOffset" matches
```

Comment:
not sure if there's any way to annotate this, but the fact that we excluded `self.freq is None` above means self[0] and self[-1] can't be NaT

Reply:
this is now anyways covered by the new NaTType overload in offsets.pyi :)